src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st

```diff
@@ -26,20 +26,38 @@ ClyBrowserMorph >> confirmEmptySystemQuery: aQuery [
 	^false
 ]
 
+{ #category : #'*Calypso-SystemTools-Core' }
+ClyBrowserMorph >> confirmEmptySystemQuery: aQuery excluding: classes [
+	
+	| showResult result |
+	
+	result := aQuery execute.
+	(result isEmpty or: [ ((result items collect: [:each | each origin]) difference: classes ) isEmpty]) ifTrue: [ ^true ].
+	
+	showResult := UIManager default confirm: 'There are '
+		, result size asString, ' ', aQuery description, '. Show them?'.
+	showResult ifTrue: [self spawnQueryBrowserOn: aQuery].
+	^false
+]
+
 { #category : #'*Calypso-SystemTools-Core' }
 ClyBrowserMorph >> confirmUnusedClasses: classes [
 	
 	| refQuery noUsers answer subclasses users |
+	
 	refQuery := ClyClassReferencesQuery toAny: classes from: self systemScope.	
-	noUsers := self confirmEmptySystemQuery: refQuery.
+	noUsers := self confirmEmptySystemQuery: refQuery excluding: classes.
 	
-	(classes anySatisfy: [ :each | each hasSubclasses ]) ifTrue: [
+	(classes anySatisfy: [ :each | |subclassesOfClass|
+		subclassesOfClass := each subclasses.
+		subclassesOfClass notEmpty and: [ (subclassesOfClass difference: classes) notEmpty ] ]) ifTrue: [
 		noUsers := false.
```

Comment:
Maybe better would be to collect all "external" subclasses and show only them. It can be like:
```
subclasses := (classes flatCollect: [:each | each subclasses]) copyWithoutAll: classes.
subclasses ifNotEmpty: [ "show them" ]
```
And same for trait users

Reply:
Ok, I'll update it